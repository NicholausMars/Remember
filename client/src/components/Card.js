import React from  'react';

/*
  @params key object
*/
// props.card is sent so i just take off a bit of it
export const Card = ({card}) => {
  return (
    <div className="container-card">
      <div className="card">
        <div className="card__side  card__side--front">
          <div className="card__title card__title--center">{card.title}</div>
        </div>
        <div className="card__side card__side--back">
          <div className="card__title card__title--border">{card.title}</div>
          <div className="card__description">
            {card.description}
          </div>
        </div>
      </div>
    </div>
  );
}

  //{/* I want a image on the front part of this card to relate to the matterial in which they want to remember*/}
