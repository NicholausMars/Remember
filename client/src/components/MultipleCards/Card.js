import React from  'react';
// import { connect } from 'react-redux';
import close from  '../../img/SVG/close-outline.svg';
import update from  '../../img/SVG/compose.svg';


/*
  props.deleteCard
  @params id
  props.updateCard
  @params card

/*
  Update: want to use the same for on the page to update the information.
  steps:
  1. click update button
  2. update send title and description to form
  3. a button icon appears close to the form field with the update logo
  4. click update then icon disappear
  5. card is updated


  needs another reducer that accept card information
*/
export const Card  = (props)=>{
  const {id, title,description} = props;
  // console.log(props);
//dummy value (works)
// const  newCard = {show:true ,id, title, description };

    return (
      <div className="container-card">
        <div className="card">
          <div className="card__side  card__side--front">
            <div className="card__title card__title--center">{title}</div>
            {/*Want the time when the card is created*/}
          </div>
          <div className="card__side card__side--back">
             <div className=" card__svg u-br-bt">
             {/*for these to happen the card must have an id value which we can pass into the function onclick*/}
             <img onClick={()=>props.updateForm({display:true ,id, title, description })}  src={update} alt='U'/>          {/*update*/}
            <img onClick={()=>props.deleteCard(id)}  src={close} alt='X'/>        {/*delete*/}
            </div>
            <div className="card__description">
              {description}
            </div>
          </div>
        </div>
      </div>
    );
  }

// update methof before refactor
// <img onClick={()=>props.updateCard(newCard)}  src={update} alt='U'/>          {/*update*/}
/*props.updateForm*/







// export default connect(null, {deleteCard})(Card);

//note:later
  //{/* I want a image on the front part of this card to relate to the matterial in which they want to remember*/}
