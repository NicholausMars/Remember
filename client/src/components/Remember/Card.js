import React from  'react';
// import { connect } from 'react-redux';
import close from  '../../img/SVG/close-outline.svg';
import update from  '../../img/SVG/compose.svg';


/*
  @params key object
  All state need to have a unique id. for Update and delete.

  2 action creators
  switch functional component into a class

  Since update changes one componets title and description
  I would have to omit that object from the state, then insert a new object with the same id.
  in order to do that i would have to have another form input of title and description with a subit button.

*/
// props.card is sent so i just take off a bit of it
// export const Card = ({title, description}) => {
// export class Card extends Component{

  // updateCard= (props) =>{
  //   console.log(props);
  // }
  // removeCard = (id) => {
  //   // console.log(id);
  //   // this.props.deleteCard(id);
  // }
export const Card  = (props)=>{
  const {id, title,description} = props;
    return (
      <div className="container-card">
        <div className="card">
          <div className="card__side  card__side--front">
            <div className="card__title card__title--center">{title}</div>
            {/*Want the time when the card is created*/}
          </div>
          <div className="card__side card__side--back">
             <div className=" card__svg u-br-bt">
             {/*for these to happen the card must have an id value which we can pass into the function onclick*/}
             <img onClick={()=>console.log('hello update')}  src={update} alt='U'/>          {/*update*/}
            <img onClick={()=>props.deleteCard(id)}  src={close} alt='X'/>        {/*delete*/}
            </div>
            <div className="card__description">
              {description}
            </div>
          </div>
        </div>
      </div>
    );
  }





// export default connect(null, {deleteCard})(Card);

//note:later
  //{/* I want a image on the front part of this card to relate to the matterial in which they want to remember*/}
