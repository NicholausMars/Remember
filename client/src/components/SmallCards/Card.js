import React from  'react';
import close from  '../../img/SVG/close-outline.svg';
import update from  '../../img/SVG/compose.svg';
export const Card  = (props)=>{
  const {id, title, description} = props;

    return (
      <div className="container-card">
        <div className="card">
          <div className="card__side  card__side--front">
            <div className="card__title card__title--center">{title}</div>
          </div>
          <div className="card__side card__side--back">
            <div className=" card__svg u-br-bt">
              <img onClick={()=> props.updateForm({display:true ,id, title, description })}  src={update} alt='U'/>
              <img onClick={()=>props.deleteCard(id)}  src={close} alt='X'/>
            </div>
            <div className="card__description">
              {description}
            </div>
          </div>
        </div>
      </div>
    );
  }






// realize i dont have to include the svg file after i already user it
// export default connect(null, {deleteCard})(Card);

//note:later
  //{/* I want a image on the front part of this card to relate to the matterial in which they want to remember*/}
