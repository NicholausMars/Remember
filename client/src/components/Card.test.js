import React from 'react';
// import { shallow, configure } from 'enzyme';
import {  mount, configure } from 'enzyme';
import  {Card}  from './Card';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

const props= {card: {title: 'KFC', description: 'The Chicken is so good'} } ;
describe('Card', ()=>{
  const card = mount(<Card {...props}/>);
  test('Card render correctly', () =>expect(card).toMatchSnapshot());
  test('render card element',()=>expect(card.find('.container-card').exists()).toBe(true));
  test('render front title',()=>expect(card.find('.card__side--front').text()).toEqual(props.card.title));
  test('render back title',()=>expect(card.find('.card__title--border').text()).toEqual(props.card.title));
  test('render back description',()=>expect(card.find('.card__description').text()).toEqual(props.card.description));
});
