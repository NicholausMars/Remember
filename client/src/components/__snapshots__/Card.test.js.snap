// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card render correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Unknown
    card={
        Object {
            "description": "The Chicken is so good",
            "title": "KFC",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "card": Object {
        "description": "The Chicken is so good",
        "title": "KFC",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="container-card"
>
        <div
                class="card"
        >
                <div
                        class="card__side  card__side--front"
                >
                        <div
                                class="card__title card__title--center"
                        >
                                KFC
                        </div>
                </div>
                <div
                        class="card__side card__side--back"
                >
                        <div
                                class="card__title card__title--border"
                        >
                                KFC
                        </div>
                        <div
                                class="card__description"
                        >
                                The Chicken is so good
                        </div>
                </div>
        </div>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card"
>
          <div
                    className="card__side  card__side--front"
          >
                    <div
                              className="card__title card__title--center"
                    >
                              KFC
                    </div>
          </div>
          <div
                    className="card__side card__side--back"
          >
                    <div
                              className="card__title card__title--border"
                    >
                              KFC
                    </div>
                    <div
                              className="card__description"
                    >
                              The Chicken is so good
                    </div>
          </div>
</div>,
        "className": "container-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="card"
>
            <div
                        class="card__side  card__side--front"
            >
                        <div
                                    class="card__title card__title--center"
                        >
                                    KFC
                        </div>
            </div>
            <div
                        class="card__side card__side--back"
            >
                        <div
                                    class="card__title card__title--border"
                        >
                                    KFC
                        </div>
                        <div
                                    class="card__description"
                        >
                                    The Chicken is so good
                        </div>
            </div>
</div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card__side  card__side--front"
>
                <div
                                className="card__title card__title--center"
                >
                                KFC
                </div>
</div>,
              <div
                className="card__side card__side--back"
>
                <div
                                className="card__title card__title--border"
                >
                                KFC
                </div>
                <div
                                className="card__description"
                >
                                The Chicken is so good
                </div>
</div>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <div
                class="card__side  card__side--front"
>
                <div
                                class="card__title card__title--center"
                >
                                KFC
                </div>
</div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="card__title card__title--center"
>
                  KFC
</div>,
                "className": "card__side  card__side--front",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <div
                    class="card__title card__title--center"
>
                    KFC
</div>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "KFC",
                    "className": "card__title card__title--center",
                  },
                  "ref": null,
                  "rendered": Array [
                    "KFC",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": <div
                class="card__side card__side--back"
>
                <div
                                class="card__title card__title--border"
                >
                                KFC
                </div>
                <div
                                class="card__description"
                >
                                The Chicken is so good
                </div>
</div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="card__title card__title--border"
>
                    KFC
</div>,
                  <div
                    className="card__description"
>
                    The Chicken is so good
</div>,
                ],
                "className": "card__side card__side--back",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <div
                    class="card__title card__title--border"
>
                    KFC
</div>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "KFC",
                    "className": "card__title card__title--border",
                  },
                  "ref": null,
                  "rendered": Array [
                    "KFC",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": <div
                    class="card__description"
>
                    The Chicken is so good
</div>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "The Chicken is so good",
                    "className": "card__description",
                  },
                  "ref": null,
                  "rendered": Array [
                    "The Chicken is so good",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "card": Object {
          "description": "The Chicken is so good",
          "title": "KFC",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="container-card"
>
          <div
                    class="card"
          >
                    <div
                              class="card__side  card__side--front"
                    >
                              <div
                                        class="card__title card__title--center"
                              >
                                        KFC
                              </div>
                    </div>
                    <div
                              class="card__side card__side--back"
                    >
                              <div
                                        class="card__title card__title--border"
                              >
                                        KFC
                              </div>
                              <div
                                        class="card__description"
                              >
                                        The Chicken is so good
                              </div>
                    </div>
          </div>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card"
>
            <div
                        className="card__side  card__side--front"
            >
                        <div
                                    className="card__title card__title--center"
                        >
                                    KFC
                        </div>
            </div>
            <div
                        className="card__side card__side--back"
            >
                        <div
                                    className="card__title card__title--border"
                        >
                                    KFC
                        </div>
                        <div
                                    className="card__description"
                        >
                                    The Chicken is so good
                        </div>
            </div>
</div>,
          "className": "container-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="card"
>
              <div
                            class="card__side  card__side--front"
              >
                            <div
                                          class="card__title card__title--center"
                            >
                                          KFC
                            </div>
              </div>
              <div
                            class="card__side card__side--back"
              >
                            <div
                                          class="card__title card__title--border"
                            >
                                          KFC
                            </div>
                            <div
                                          class="card__description"
                            >
                                          The Chicken is so good
                            </div>
              </div>
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card__side  card__side--front"
>
                  <div
                                    className="card__title card__title--center"
                  >
                                    KFC
                  </div>
</div>,
                <div
                  className="card__side card__side--back"
>
                  <div
                                    className="card__title card__title--border"
                  >
                                    KFC
                  </div>
                  <div
                                    className="card__description"
                  >
                                    The Chicken is so good
                  </div>
</div>,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <div
                  class="card__side  card__side--front"
>
                  <div
                                    class="card__title card__title--center"
                  >
                                    KFC
                  </div>
</div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="card__title card__title--center"
>
                    KFC
</div>,
                  "className": "card__side  card__side--front",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <div
                      class="card__title card__title--center"
>
                      KFC
</div>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "KFC",
                      "className": "card__title card__title--center",
                    },
                    "ref": null,
                    "rendered": Array [
                      "KFC",
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": <div
                  class="card__side card__side--back"
>
                  <div
                                    class="card__title card__title--border"
                  >
                                    KFC
                  </div>
                  <div
                                    class="card__description"
                  >
                                    The Chicken is so good
                  </div>
</div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="card__title card__title--border"
>
                      KFC
</div>,
                    <div
                      className="card__description"
>
                      The Chicken is so good
</div>,
                  ],
                  "className": "card__side card__side--back",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <div
                      class="card__title card__title--border"
>
                      KFC
</div>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "KFC",
                      "className": "card__title card__title--border",
                    },
                    "ref": null,
                    "rendered": Array [
                      "KFC",
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": <div
                      class="card__description"
>
                      The Chicken is so good
</div>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "The Chicken is so good",
                      "className": "card__description",
                    },
                    "ref": null,
                    "rendered": Array [
                      "The Chicken is so good",
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
